parlai/agents/rag/rag.py

```diff
@@ -553,6 +553,18 @@ def _set_text_vec(
         """
         return self._generation_agent._set_text_vec(self, obs, history, truncate)
 
+    def _add_generation_metrics(self, batch: Batch, preds: List[torch.Tensor]):
```

Comment:
won't the RagAgent inherit this from the TGA?

Reply:
i needed to make a change to not use `batch.batchsize`, since that fails for RAG Sequence.

I suppose the change could actually just be made at the TGA level